From: Jonathan Nieder <jrnieder@gmail.com>
Date: Tue, 29 Sep 2009 21:40:37 -0500
Subject: [PATCH 4/9] Add decompress_cmd() for running gzip-like commands

Factor out a function that can be used to invoke gzip -dc,
bzip2 -dc, or lzma -dc.

This does not make the file any shorter (it actually makes it
longer).  The only reason for this function to exist is symmetry
with respect to compress_cmd().
---
 lib/dpkg/compression.h |   12 +++++++++---
 1 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/lib/dpkg/compression.h b/lib/dpkg/compression.h
index 9205a6d..f5cff3f 100644
--- a/lib/dpkg/compression.h
+++ b/lib/dpkg/compression.h
@@ -49,6 +49,12 @@ static void compress_cmd(int fd_in, int fd_out, const char *file,
 	fd_fd_filter(fd_in, fd_out, file, cmd, combuf, desc);
 }
 
+static void decompress_cmd(int fd_in, int fd_out, const char *file,
+                           const char *cmd, const char *desc)
+{
+	fd_fd_filter(fd_in, fd_out, file, cmd, "-dc", desc);
+}
+
 static void decompress_noop(int fd_in, int fd_out, const char *desc)
 {
 	fd_fd_copy(fd_in, fd_out, -1, _("%s: decompression"), desc);
@@ -117,7 +123,7 @@ static void compress_gzip(int fd_in, int fd_out, char compression,
 #else /* !WITH_ZLIB */
 static void decompress_gzip(int fd_in, int fd_out, const char *desc)
 {
-  fd_fd_filter(fd_in, fd_out, GZIP, "gzip", "-dc", desc);
+	decompress_cmd(fd_in, fd_out, GZIP, "gzip", desc);
 }
 
 static void compress_gzip(int fd_in, int fd_out, char compression,
@@ -182,7 +188,7 @@ static void compress_bzip2(int fd_in, int fd_out, char compression,
 #else /* !WITH_BZ2 */
 static void decompress_bzip2(int fd_in, int fd_out, const char *desc)
 {
-  fd_fd_filter(fd_in, fd_out, BZIP2, "bzip2", "-dc", desc);
+	decompress_cmd(fd_in, fd_out, BZIP2, "bzip2", desc);
 }
 
 static void compress_bzip2(int fd_in, int fd_out, char compression,
@@ -194,7 +200,7 @@ static void compress_bzip2(int fd_in, int fd_out, char compression,
 
 static void decompress_lzma(int fd_in, int fd_out, const char *desc)
 {
-	fd_fd_filter(fd_in, fd_out, LZMA, "lzma", "-dc", desc);
+	decompress_cmd(fd_in, fd_out, LZMA, "lzma", desc);
 }
 
 static void compress_lzma(int fd_in, int fd_out, char compression,
-- 
1.6.5.rc1.199.g596ec

